pkg/utils/utils.go

```diff
@@ -282,22 +282,48 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 				if sriovnetworkv1.IndexInRange(vfID, group.VfRange) {
 					isRdma = group.IsRdma
 					if sriovnetworkv1.StringInArray(group.DeviceType, DpdkDrivers) {
-						driver = group.DeviceType
+						dpdkDriver = group.DeviceType
 					}
 					break
 				}
 			}
-			if strings.EqualFold(iface.LinkType, "IB") {
-				if err = setVfGuid(addr, pfLink); err != nil {
-					return err
+
+			// only set GUID and MAC for VF with default driver
+			// for userspace drivers like vfio we configure the vf mac using the kernel nic mac address
+			// before we switch to the userspace driver
+			if yes, d := hasDriver(addr); yes && !sriovnetworkv1.StringInArray(d, DpdkDrivers) {
```

Comment:
/cc @adrianchiris @mskrocki @pliurh @zshi-redhat 

I add this validation instead of `if iface.NumVfs != ifaceStatus.NumVfs ` that was proposed under https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/245.

I think is a better solution because we are inside a for loop going over all the vfs. if there is an issue with one of the vfs(for example the intel driver get stuck) we exist the `configSriovDevice` function with an error. Then in the next reconcile the iface.NumVfs will be equal to the ifaceStatus.NumVfs because the vfs got already created here (https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/233/files#diff-81ddbadfb415ccbb9c7af84f11668d1aa5e53c34025bf86d4702f16b4e42f045R246) but we didn't really finish allocating the GUID or the mac address to all the vfs.